src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -171,18 +171,32 @@ private function dumpArrayList(string $output, string $input, string $contextPro
         $memberShape = $shape->getMember()->getShape();
 
         return strtr('
-            $index = -1;
-            foreach (INPUT as $listValue) {
-                $index++;
-                MEMBER_CODE
+            if (null !== INPUT) {
```

Comment:
This not needed (or not the right implementation). None of other `dump*` method performs such check because, this should have been done before.

Reply:
Got it, you should remove at line 62
```diff
-if ($shape instanceof ListShape || $shape instanceof MapShape) {
+if ($shape instanceof MapShape) {
   $body = 'MEMBER_CODE';
   $inputElement = '$this->' . $member->getName();
 }